tests/main.test.ts

```diff
@@ -1,22 +1,307 @@
+        "https://api.github.com/repos/:org/:repo/pulls/:id/reviews",
+        () => {
+          return HttpResponse.json([{ state: "APPROVED" }, { state: "APPROVED" }]);
+        },
+        { once: true }
+      )
+    );
+    jest.mock(actionsGithubPackage, () => ({
+      context: {
+        repo: {
+          owner: {
+            login: "ubiquibot",
+          },
+        },
+        workflow,
+        payload: {
+          inputs: {
+            eventName: "pull_request.closed",
+            settings: JSON.stringify({
+              databaseUrl: dbName,
+            }),
+            eventPayload: JSON.stringify({
+              pull_request: {
+                html_url: htmlUrl,
+              },
+            }),
+            env: {
+              workflowName: workflow,
+            },
+          },
+        },
+      },
+    }));
+    const run = (await import("../src/action")).run;
+    await expect(run()).resolves.toMatchObject({ status: 200 });
+    const pullRequests = await dataSource.getRepository(PullRequest).find();
+    expect(pullRequests).toHaveLength(0);
+  });
+
+  it("Should not close a PR that is not past the threshold", async () => {
+    const dataSource = await initializeDataSource(dbName);
+    const pr = new PullRequest();
+    pr.url = htmlUrl;
+    pr.lastActivity = new Date();
+    await pr.save();
+    server.use(
+      http.get(
+        "https://api.github.com/repos/:org/:repo/pulls/:id/merge",
+        () => {
+          return HttpResponse.json({}, { status: 404 });
+        },
+        { once: true }
+      ),
+      http.get(
+        "https://api.github.com/repos/:org/:repo/issues/:id/timeline",
+        () => {
+          return HttpResponse.json([{ id: 1, created_at: new Date() }]);
+        },
+        { once: true }
+      )
+    );
+    jest.mock(actionsGithubPackage, () => ({
+      context: {
+        repo: {
+          owner: {
+            login: "ubiquibot",
+          },
+        },
+        workflow,
+        payload: {
+          inputs: {
+            eventName: "push",
+            settings: JSON.stringify({
+              databaseUrl: dbName,
+            }),
+            eventPayload: JSON.stringify({
+              pull_request: {
+                html_url: htmlUrl,
+              },
+            }),
+            env: {
+              workflowName: workflow,
+            },
+          },
+        },
+      },
+    }));
+    const run = (await import("../src/action")).run;
+    await expect(run()).resolves.toMatchObject({ status: 200 });
+    const pullRequests = await dataSource.getRepository(PullRequest).find();
+    expect(pullRequests).toHaveLength(1);
+  });
+
+  it("Should close a PR that is past the threshold", async () => {
+    const dataSource = await initializeDataSource(dbName);
+    const pr = new PullRequest();
+    pr.url = htmlUrl;
+    const lastActivityDate = new Date();
+    lastActivityDate.setDate(new Date().getDate() - 8);
+    pr.lastActivity = lastActivityDate;
```

Comment:
and then how does this test pass if `lastActivity` is not read anywhere?

Reply:
Removed within https://github.com/ubiquibot/automated-merging/pull/1/commits/656bcbc1b4c7841c831366d0aa86afa6638cfcd0